### ISSUE TITLE
After Remote Upsert

### ISSUE THREAD

**Issue by anouarkacem** [strongloop/loopback#3805]
I was trying to PATCH (upsert) some model and catch the result with the AfterRemote Hook, I tried with the code below, nothing happens, no response no errors even with the afterRemoteError 

```
 model.afterRemote('upsert', function(ctx, result, next) {
  console.log(result);
  next();
 })
```
Replacing the method name with **updateOrCreate** doesn't work, no response also, however it works with this method name **patchOrCreate** I found it under [loopback/lib/persisted-model.js line 119](https://github.com/strongloop/loopback/blob/master/lib/persisted-model.js)

And in same file, line 684 `setRemoting(PersistedModel, 'patchOrCreate', upsertOptions);` then just changed it to **upsert** and it works. 

I am not sure if it is a bug or not.

Why even change its name, after all **patchOrCreate** was mentioned in the documentation once [here](https://loopback.io/doc/en/lb3/3.0-Release-Notes.html) 

Can anyone of add this to the documentation or just add the upsert method in the setRemoting, thank you.

**Comment by bajtos** [comment]
Let me rephrase your report as I understand it: `afterRemote('patchOrCreate')` works, but it does not work with method names "updateOrCreate" and "upsert". Is that correct?

Such behavior looks like a bug in the implementation of our remote hooks to me - perhaps the hooks are not recognizing aliases? This particular method is registered as "patchOrCreate" with two aliases "updateOrCreate" and "upsert".

>  just add the upsert method in the setRemoting

This is already happening, see
https://github.com/strongloop/loopback/blob/6ddf268cb626521971f5e5b63382bf08af78285a/lib/persisted-model.js#L666

Here is the code responsible for registering "afterRemote" hooks in strong-remoting, it's a good place where to start looking: https://github.com/strongloop/strong-remoting/blob/c937334dbf6d8b69d61171a3e2ba29df966c61c9/lib/remote-objects.js#L370-L372
